exceptions.hpp

```diff
@@ -0,0 +1,13 @@
+#include "exception"
+
+class RearGearChangeException : public std::exception {
+  const char* what () const throw (){
```

Comment:
```suggestion
  const char* what () const {
```
do not use `throw()` specifier. It was removed in C++17.

BTW it's better not to overload what() function, but just write a constructor that passes this message to a base class constructor. See others solutions.

Reply:
Changed to class constructor and inheritance to std::logic_exception